src/Vault.sol

```diff
@@ -126,26 +126,26 @@ contract Vault is IVault, VaultToken, Ownable {
         SettlementGuard.accountDelta(msg.sender, currency, -(paid.toInt128()));
     }
 
-    function settleAndRefund(Currency currency, address to)
+    function settleAndMintRefund(Currency currency, address to)
         external
         payable
         override
         isLocked
         returns (uint256 paid, uint256 refund)
     {
-        paid = currency.balanceOfSelf() - reservesOfVault[currency];
-        int256 currentDelta = SettlementGuard.getCurrencyDelta(msg.sender, currency);
+        uint256 reservesBefore = reservesOfVault[currency];
+        reservesOfVault[currency] = currency.balanceOfSelf();
+        paid = reservesOfVault[currency] - reservesBefore;
```

Comment:
For this part, the function would revert if reservesBefore is larger than the current reserve, since it would mean that the locker did not transfer anything to increase the current balance right?

Reply:
if the locker didn't transfer anything, this function shouldn't revert like the existing `settle()`.

Can think of 1 scenario for revert: 
1/ might be for rebasing token, where a negative rebase was done which cause the current balance in vault to be lesser than reserveBefore

cc @chefburger or @ChefSnoopy if can add on to more scenarios where revert might happen

Reply:
I cant come with other cases either, but i think it's fine to just let it revert if reservesBefore is greater than current.